### PR TITLE
Fix some swig warnings and fix the issue that Node::mobility is not bound, update swig to 4.1.0 release

### PR DESCRIPTION
**Pull request by dumganhar**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/tools/swig-config/genbindings.js [hunk 1]
```diff
@@ -232,8 +232,13 @@ function generateBindings(swigArgs, interfaceFile, generatedCppFile) {
         const ret = spawnSync(SWIG_EXE, swigArgs, {
             stdio: ['ignore', process.stdout, process.stderr],
         });
-        if (ret.status !== 0) {
-            process.exit(ret.status);
+        let retCode = ret.status;
+        if (retCode !== 0) {
+            console.error(`==> ERROR: spawnSync returns ${retCode}`);
+            if (typeof retCode !== 'number') {
```
Comment (dumganhar) [inline native/tools/swig-config/genbindings.js:238]: While swig crashes, retCode may be `null` which is not a number.

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/tools/swig-config/genbindings.js (hunk 1 above)
Comment (minggo) [reply native/tools/swig-config/genbindings.js:238]: I think it is better to add a comment.

File: native/tools/swig-config/scene.i [hunk 2]
```diff
@@ -284,7 +285,7 @@ using namespace cc;
 %attribute_writeonly(cc::Node, Mat4&, matrix, setMatrix);
 %attribute(cc::Node, uint32_t, hasChangedFlags, getChangedFlags, setChangedFlags);
 %attribute(cc::Node, bool, _persistNode, isPersistNode, setPersistNode);
-%attribute(cc::Node, cc::MobilityMode, _mobility, getMobility, setMobility);
+%attribute(cc::Node, cc::MobilityMode, mobility, getMobility, setMobility);
```
Comment (minggo) [inline native/tools/swig-config/scene.i:288]: Why change it? I don't see any modification of the implementation.

Comment (dumganhar) [reply native/tools/swig-config/scene.i:288]: https://github.com/cocos/cocos-engine/blob/develop/cocos/scene-graph/node.ts#L1679

To keep the same logic as web.

_mobilty is a public property that will be automatically bound, no need to use an attribute directive to declare it. 

The old code will  not bind `mobility` property to JS.

**Review by minggo — APPROVED** [review]

**Pull request merged**